clients/src/test/java/org/apache/kafka/test/MockMetricsReporter.java

```diff
@@ -26,6 +27,7 @@
 public class MockMetricsReporter implements MetricsReporter {
     public static final AtomicInteger INIT_COUNT = new AtomicInteger(0);
     public static final AtomicInteger CLOSE_COUNT = new AtomicInteger(0);
+    public static String clientId;
```

Comment:
This should not be static.  We might want to run unit tests in parallel.

Reply:
Good point. I have pushed a new commit that addresses this.

We can retrieve the `MockMetricsReporter` object from the `metrics` object that is already exposed for testing. The caveat is that we need to iterate through the `reporters` list to fetch the `MockMetricsReporter` object (there is also a `JmxReporter` object in the list).